docs/concepts/controllers/statefulsets.md

```diff
@@ -0,0 +1,171 @@
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+        volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html
+  volumeClaimTemplates:
+  - metadata:
+      name: www
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+### Pod Identity
+StatefulSet Pods have a unique identity that is comprised of an ordinal, a 
+stable network identity, and stable storage. The identity sticks to the Pod, 
+regardless of which node it's (re) scheduled on.
+
+__Ordinal Index__
+
+For a StatefulSet with N replicas, each Pod in the StatefulSet will be 
+assigned an integer ordinal, in the range [0,N), that is unique over the Set. 
+
+__Stable Network ID__
+
+The hostname of a Pod in a StatefulSet is derived from the name of the 
+StatefulSet and the ordinal of the Pod. The pattern for the constructed hostname 
+is `$(statefulset name)-$(ordinal)`. The example above will create three Pods 
+named `web-0,web-1,web-2`.
+A StatefulSet can use a [Headless Service](/docs/user-guide/services/#headless-services)
+to control the domain of its Pods. The domain managed by this Service takes the form: 
+`$(service name).$(namespace).svc.cluster.local`, where "cluster.local" 
+is the [cluster domain](http://releases.k8s.io/{{page.githubbranch}}/build/kube-dns/README.md#how-do-i-configure-it). 
+As each Pod is created, it gets a matching DNS subdomain, taking the form: 
+`$(podname).$(governing service domain)`, where the governing service is defined 
+by the `serviceName` field on the StatefulSet.
+
+Here are some examples of choices for Cluster Domain, Service name, 
+StatefulSet name, and how that affects the DNS names for the StatefulSet's Pods.
+
+Cluster Domain | Service (ns/name) | StatefulSet (ns/name)  | StatefulSet Domain  | Pod DNS | Pod Hostname |
+-------------- | ----------------- | ----------------- | -------------- | ------- | ------------ |
+ cluster.local | default/nginx     | default/web       | nginx.default.svc.cluster.local | web-{0..N-1}.nginx.default.svc.cluster.local | web-{0..N-1} |
+ cluster.local | foo/nginx         | foo/web           | nginx.foo.svc.cluster.local     | web-{0..N-1}.nginx.foo.svc.cluster.local     | web-{0..N-1} |
+ kube.local    | foo/nginx         | foo/web           | nginx.foo.svc.kube.local        | web-{0..N-1}.nginx.foo.svc.kube.local        | web-{0..N-1} |
+
+Note that Cluster Domain will be set to `cluster.local` unless 
+[otherwise configured](http://releases.k8s.io/{{page.githubbranch}}/build/kube-dns/README.md#how-do-i-configure-it).
+
+__Stable Storage__
+
+[PersistentVolumes](/docs/user-guide/volumes/), one for each Volume Claim Template, 
```

Comment:
"Kubernetes creates one [PersistentVolume] for each Volume Claim Template, as specified in the StatefulSet's `volumeClaimTemplates` field."